src/main/java/calculation/CalculateCoordinatesInterpolationAzimuthLength.java

```diff
@@ -0,0 +1,92 @@
+package calculation;
+
+import model.Location;
+import model.LocationPoint;
+import org.gavaghan.geodesy.Ellipsoid;
+import org.gavaghan.geodesy.GeodeticCalculator;
+import org.gavaghan.geodesy.GlobalCoordinates;
+import utils.Utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculateCoordinatesInterpolationAzimuthLength {
+  public List<Location> calculatePointsOnRoutePrototype(LocationPoint locationPoint) {
+    Location location;
+    List<Location> resultsCoordinates = new ArrayList<>();
+    List<Double> newCalculatedLat = new ArrayList<>();
+    List<Double> newCalculatedLong = new ArrayList<>();
+    newCalculatedLat.add(locationPoint.getPointLatitude().get(0));
+    newCalculatedLong.add(locationPoint.getPointLongitude().get(0));
+    List<Integer> lastIndexUsed = new ArrayList<>();
+    lastIndexUsed.add(0);
+
+    double step = locationPoint.getStep();
+    int locationIndex = locationPoint.getPointDistance().size();
+    for (int oldPosition = 0; oldPosition < locationIndex; oldPosition++) {
+      double oldPositionLength = locationPoint.getPointDistance().get(oldPosition);
+      double oldPositionAzimuth = locationPoint.getPointAzimuth().get(oldPosition);
+      GeodeticCalculator geoCalc = new GeodeticCalculator();
```

Comment:
do you need to create the geoCalc object in every loop step? or you can move it before the for loop?